File: Cocoa/DeveloperTools/Timeline/YJTimeline.m
//
//  YJTimeline.m
//  YJCocoa
//
//  HomePage:https://github.com/937447974/YJCocoa
//  YJ技术支持群:557445088
//
//  Created by 阳君 on 2019/1/19.
//  Copyright © 2016年 YJCocoa. All rights reserved.
//

#import "YJTimeline.h"
#import "YJNSLog.h"

@implementation YJTimeline
static NSMutableString *YJTimeline_log;
CFAbsoluteTime YJTimeline_time;

+ (void)addSteps:(NSString *)steps {
    if (!YJTimeline_log) YJTimeline_log = NSMutableString.string;
    CFAbsoluteTime currentTime = CFAbsoluteTimeGetCurrent();
    CFAbsoluteTime timeInterval = currentTime - YJTimeline_time;
    YJTimeline_time = currentTime;
    if (YJTimeline_log.length) [YJTimeline_log appendFormat:@" %f", timeInterval];
    [YJTimeline_log appendFormat:@"\n\t%@", steps];
}

+ (void)end {
    YJLogDebug(YJTimeline_log);
    YJTimeline_log = nil;
}

@end

File: YJCocoa/Classes/DeveloperTools/Timeline/YJTimeline.swift
//
//  YJTimeline.swift
//  YJCocoa
//
//  HomePage:https://github.com/937447974/YJCocoa
//  YJ技术支持群:557445088
//
//  Created by 阳君 on 2019/6/18.
//  Copyright © 2016-现在 YJCocoa. All rights reserved.
//

import UIKit

/// 时间轴记录
@objcMembers
public class YJTimeline: NSObject {
    
    static var log: String?
    static var time: CFAbsoluteTime = 0
    
    /// 添加时间轴步骤
    public static func add(step: String) {
        if self.log == nil {
            self.log = step
        } else {
            let time = String(format: " %.3f", CFAbsoluteTimeGetCurrent() - self.time)
            self.log = self.log! + time + "\n\t" + step
        }
        self.time = CFAbsoluteTimeGetCurrent()
    }
    
    /// 结束并打印日志
    public static func end() {
        guard let log = self.log else {
            return
        }
        let time = String(format: " %.3f", CFAbsoluteTimeGetCurrent() - self.time)
        YJLogDebug(log + time)
        self.log = nil
    }
    
}
